### PR TITLE
Add checkstyle rule for test setup/teardown method names

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

File: gradle-baseline-java-config/resources/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -307,6 +307,10 @@
             <property name="format" value="(\/\/TODO)|(\/\/ TODO:)|(\/\/ TODO\(\))|(\/\/ TODO\(.*\):[^ ])|(\/\/ TODO\(.*\)[^:])"/>
             <property name="message" value="TODO format: // TODO(flastname): explanation"/>
         </module>
+        <module name="RegexpSinglelineJava">
+            <property name="format" value="(void setUp\(\))|(void setup\(\))|(void setupStatic\(\))|(void setUpStatic\(\))|(void beforeTest\(\))|(void teardown\(\))|(void tearDown\(\))"/>
```
Comment (mnazbro) [inline gradle-baseline-java-config/resources/checkstyle/checkstyle.xml]: Why not just enforce one of `setUp` and `setup`?

**Comment by uschi2000** [comment]
It's inverse, this rule throws up when it sees one of those patterns.

File: gradle-baseline-java-config/resources/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -307,6 +307,10 @@
...
+            <property name="message" value="Test setup/teardown methods are called before(), beforeStatic(), after(), afterStatic(), but not setUp, teardown, etc."/>
```
Comment (ellisjoe) [inline gradle-baseline-java-config/resources/checkstyle/checkstyle.xml]: `beforeClass()` and `afterClass()`?

**Comment by uschi2000** [comment]
Ah,yes. Thanks

**Comment by schlosna** [comment]
Honestly curious what value this adds? Specifically I'm thinking about case of having multiple independent @After that release resources and allow JUnit to handle ensuring shutdown appropriately. I guess you could have each one be a @Rule, but seems heavyweight and friction for little gain.

**Comment by uschi2000** [comment]
I haven't seen a class with multiple `@After`, do you have an example?
The benefits of this change are: it doesn't matter what you call this method, so let's agree on one canonical name for the 99% cases (`before()`) so that we don't need to argue on PRs and can find the setup method quickly.

**Comment by jmcampanini** [comment]
+1. instead of multiple `@After` methods, could you put them in a single one? it would also disambiguate what order those methods are ran in, since order isn't guaranteed by junit

**Pull request merged**